packages/e2e-tests/specs/widgets/editing-widgets.test.js

```diff
@@ -394,109 +394,116 @@ describe( 'Widgets screen', () => {
 	` );
 	} );
 
-	async function addMarquee() {
-		// There will be 2 matches here.
-		// One is the in-between inserter,
-		// and the other one is the button block appender.
-		const [ inlineInserterButton ] = await findAll( {
-			role: 'combobox',
-			name: 'Add block',
-		} );
-		await inlineInserterButton.click();
+	describe( 'Function widgets', () => {
+		async function addMarquee() {
+			// There will be 2 matches here.
+			// One is the in-between inserter,
+			// and the other one is the button block appender.
+			const [ inlineInserterButton ] = await findAll( {
+				role: 'combobox',
+				name: 'Add block',
+			} );
+			await inlineInserterButton.click();
 
-		// TODO: Convert to find() API from puppeteer-testing-library.
-		const inserterSearchBox = await page.waitForSelector(
-			'aria/Search for blocks and patterns[role="searchbox"]'
-		);
-		await expect( inserterSearchBox ).toHaveFocus();
+			// TODO: Convert to find() API from puppeteer-testing-library.
+			const inserterSearchBox = await page.waitForSelector(
+				'aria/Search for blocks and patterns[role="searchbox"]'
+			);
+			await expect( inserterSearchBox ).toHaveFocus();
 
-		await page.keyboard.type( 'Marquee' );
+			await page.keyboard.type( 'Marquee' );
 
-		const inlineQuickInserter = await find( {
-			role: 'listbox',
-			name: 'Blocks',
-		} );
-		const marqueeBlockOption = await find(
-			{
-				role: 'option',
-			},
-			{
-				root: inlineQuickInserter,
-			}
-		);
-		await marqueeBlockOption.click();
-	}
+			const inlineQuickInserter = await find( {
+				role: 'listbox',
+				name: 'Blocks',
+			} );
+			const marqueeBlockOption = await find(
+				{
+					role: 'option',
+				},
+				{
+					root: inlineQuickInserter,
+				}
+			);
+			await marqueeBlockOption.click();
+		}
 
-	it( 'Should add and save the marquee widget', async () => {
-		await activatePlugin( 'gutenberg-test-marquee-widget' );
-		await visitAdminPage( 'widgets.php' );
+		afterEach( async () => {
+			await deleteAllWidgets();
+			await deactivatePlugin( 'gutenberg-test-marquee-widget' );
+		} );
 
-		await addMarquee();
+		it( 'Should add and save the marquee widget', async () => {
+			await activatePlugin( 'gutenberg-test-marquee-widget' );
```

Comment:
How about moving this into a newly created `beforeEach()` (so setup and teardown are symmetrical)? Or better yet maybe, move this one into `beforeAll(), and change the `afterEach()` into an `afterAll()`? This would allow adding other tests to the 'Function widgets' group later and would only activate the plugin once at the very beginning, and deactivated it at the very end.

Reply:
That's a good note, thank you! I went with `beforeEach` and `afterEach` for now, let's see if the tests pass before tweaking that further, I am worried that calling `deleteAllWidgets` without deactivating and reactivating the plugin could cause the tests to fail.